Source/santasyncservice/SNTSyncFCM.m

```diff
@@ -212,24 +203,13 @@ - (void)reachabilityRestored {
 
 /**  Start listening for network state changes on a background thread. */
 - (void)startReachability {
-  if (self.reachability) return;
   LOGD(@"Reachability started.");
-  self.reachability =
-    SCNetworkReachabilityCreateWithName(kCFAllocatorDefault, _connectComponents.host.UTF8String);
-  SCNetworkReachabilityContext context = {.info = (__bridge void *)self};
-  if (SCNetworkReachabilitySetCallback(self.reachability, reachabilityHandler, &context)) {
-    SCNetworkReachabilitySetDispatchQueue(
-      self.reachability, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0));
-  }
+  nw_path_monitor_start(self.pathMonitor);
 }
 
 /**  Stop listening for network state changes. */
 - (void)stopReachability {
-  if (self.reachability) {
-    SCNetworkReachabilitySetDispatchQueue(self.reachability, NULL);
-    if (self.reachability) CFRelease(self.reachability);
-    self.reachability = NULL;
-  }
+  nw_path_monitor_cancel(self.pathMonitor);
```

Comment:
Same Q here.

Reply:
This is also undocumented and the behavior is also weird. It seems that if you "overcancel" the monitor gets broken and can't be started. I've changed this too.